STM32 BxCAN have integer overflow in expression

**Describe the bug**
On nearly all high-speed clocked boards based on STM32 the CAN driver test suite will fail because of warnings will become errors. Core of the problem is a wrong integer DIV/MUL arithmetic in `drivers/can/can_stm32_bxcan.c` for `#define CAN_INIT_TIMEOUT (10 * sys_clock_hw_cycles_per_sec() / MSEC_PER_SEC)`.

The intention of this CPP define is the pre-calculation of exact amount of kernel cycles that match 10 milliseconds depending on the board clock setup. The first term leads to a really big number on boards with high-speed clocking – the overflow. Better her would be to write:

```
#define CAN_INIT_TIMEOUT (10 * (sys_clock_hw_cycles_per_sec() / MSEC_PER_SEC))
```

That means divide first (to little numbers) and multiply then by 10 to get cycles for 10 milliseconds.

**To Reproduce**

Here is an example console log:

```
west twister --verbose --inline-logs \
             --platform nucleo_f767zi \
             --test tests/drivers/can/shell/drivers.can.shell
```

…breaks with (CMake/gcc output in build.log):

```
/.../workspace/zephyr/drivers/can/can_stm32_bxcan.c: In function 'can_stm32_enter_init_mode':
/.../workspace/zephyr/drivers/can/can_stm32_bxcan.c:24:31: error: integer overflow in expression of type 'int' results in '-2134967296' [-Werror=overflow]
   24 | #define CAN_INIT_TIMEOUT  (10 * sys_clock_hw_cycles_per_sec() / MSEC_PER_SEC)
      |                               ^
/.../workspace/zephyr/drivers/can/can_stm32_bxcan.c:361:53: note: in expansion of macro 'CAN_INIT_TIMEOUT'
  361 |                 if (k_cycle_get_32() - start_time > CAN_INIT_TIMEOUT) {
      |                                                     ^~~~~~~~~~~~~~~~
```

Same error for `can_stm32_leave_init_mode()` and `can_stm32_leave_sleep_mode()`.

**Expected behavior**
Twister integration tests without warnings (errors).

**Impact**
Showstopper for proper integration tests.

**Logs and console output**
See above.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.16.1
 - main branch, commit SHA f8e812aa3f5ace9abb01affe2aed267f2a6cde08

**Additional context**
Non.

--- a/drivers/can/can_stm32_bxcan.c
+++ b/drivers/can/can_stm32_bxcan.c
@@ -21,7 +21,7 @@
 
 LOG_MODULE_REGISTER(can_stm32, CONFIG_CAN_LOG_LEVEL);
 
-#define CAN_INIT_TIMEOUT  (10 * sys_clock_hw_cycles_per_sec() / MSEC_PER_SEC)
+#define CAN_INIT_TIMEOUT  (10 * (sys_clock_hw_cycles_per_sec() / MSEC_PER_SEC))
 
 #define DT_DRV_COMPAT st_stm32_bxcan
 

